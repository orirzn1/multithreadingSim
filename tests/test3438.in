L15
S6
O13
N10

T0
I@0x00000000
ADD $0, $2, $7
ADDI $4, $4, 0x4
SUBI $5, $5, 0x4
ADDI $4, $1, 0x2
ADD $0, $6, $1
SUBI $3, $4, 0x2
ADDI $4, $1, 0x2
HALT $0

T1
I@0x00000000
SUBI $7, $0, 0x1
ADDI $1, $7, 0x3
STORE $2, $5, $4
ADD $5, $5, $2
HALT $0

T2
I@0x00000000
ADD $3, $6, $4
ADD $7, $2, $1
ADDI $3, $6, 0x2
ADDI $6, $5, 0x2
ADDI $3, $4, 0x2
SUB $7, $2, $1
SUBI $3, $6, 0x1
HALT $0

T3
I@0x00000000
SUB $3, $2, $3
STORE $3, $6, $3
LOAD $6, $6, 0x3
STORE $3, $7, 0x4
SUB $3, $2, $7
HALT $0

T4
I@0x00000000
ADD $3, $3, $5
ADD $3, $4, $7
SUBI $0, $0, 0x1
STORE $5, $5, 0x2
SUBI $2, $6, 0x2
ADD $6, $3, $1
HALT $0

T5
I@0x00000000
STORE $7, $4, 0x0
SUB $3, $4, $5
LOAD $4, $1, 0x0
ADD $7, $0, $3
STORE $7, $7, $4
STORE $5, $4, $0
SUB $2, $7, $7
SUB $3, $3, $2
LOAD $5, $4, 0x2
HALT $0

T6
I@0x00000000
SUB $2, $1, $7
SUB $4, $1, $6
LOAD $6, $7, 0x4
ADD $2, $7, $3
SUB $0, $5, $2
LOAD $4, $3, $6
LOAD $5, $4, $4
STORE $6, $2, $2
STORE $3, $0, $5
HALT $0

T7
I@0x00000000
STORE $1, $6, 0x0
SUB $3, $2, $2
SUBI $7, $3, 0x2
ADDI $5, $7, 0x1
HALT $0

T8
I@0x00000000
STORE $1, $7, 0x1
ADDI $1, $1, 0x3
ADDI $7, $2, 0x2
SUB $5, $3, $5
SUBI $4, $0, 0x4
STORE $7, $3, $7
HALT $0

T9
I@0x00000000
ADDI $0, $1, 0x1
ADD $3, $1, $5
STORE $3, $1, 0x2
SUB $6, $0, $1
ADDI $7, $5, 0x1
STORE $0, $1, $3
HALT $0

D@0x0
0x8
0x2
0x9
0x5
0x6
0xA
0x6
0x7
0x8
