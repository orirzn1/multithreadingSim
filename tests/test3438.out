
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x2	R5 = 0xFFFFFFFC	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0x0	R1 = 0x2	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFF
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x1	R4 = 0x0	R5 = 0x0	R6 = 0x2	R7 = 0x0
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 4:
	R0 = 0xFFFFFFFF	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x2	R7 = 0x0
Register file thread id 7:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFF	R6 = 0x0	R7 = 0xFFFFFFFE
Register file thread id 8:
	R0 = 0x0	R1 = 0x3	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0x2
Register file thread id 9:
	R0 = 0x1	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x1	R7 = 0x1
Blocked MT CPI for this program 6.178082

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x2	R5 = 0xFFFFFFFC	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0x0	R1 = 0x2	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFF
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x1	R4 = 0x0	R5 = 0x0	R6 = 0x2	R7 = 0x0
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 4:
	R0 = 0xFFFFFFFF	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 7:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFF	R6 = 0x0	R7 = 0xFFFFFFFE
Register file thread id 8:
	R0 = 0x0	R1 = 0x3	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0x2
Register file thread id 9:
	R0 = 0x1	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x1	R7 = 0x1
Finegrained Multithreading CPI for this program 1.465753

